pkg/network/payload/version.go

```diff
@@ -26,60 +14,53 @@ type Version struct {
 	Magic config.NetMode
 	// currently the version of the protocol is 0
 	Version uint32
-	// currently 1
-	Services uint64
 	// timestamp
 	Timestamp uint32
-	// port this server is listening on
-	Port uint16
 	// it's used to distinguish the node from public IP
 	Nonce uint32
 	// client id
 	UserAgent []byte
-	// Height of the block chain
-	StartHeight uint32
-	// Whether to receive and forward
-	Relay bool
+	// List of available network services
+	Capabilities []capability.Capability
 }
 
 // NewVersion returns a pointer to a Version payload.
-func NewVersion(magic config.NetMode, id uint32, p uint16, ua string, h uint32, r bool) *Version {
+func NewVersion(magic config.NetMode, id uint32, ua string, c []capability.Capability) *Version {
 	return &Version{
-		Magic:       magic,
-		Version:     0,
-		Services:    nodePeerService,
-		Timestamp:   uint32(time.Now().UTC().Unix()),
-		Port:        p,
-		Nonce:       id,
-		UserAgent:   []byte(ua),
-		StartHeight: h,
-		Relay:       r,
+		Magic:        magic,
+		Version:      0,
+		Timestamp:    uint32(time.Now().UTC().Unix()),
+		Nonce:        id,
+		UserAgent:    []byte(ua),
+		Capabilities: c,
 	}
 }
 
 // DecodeBinary implements Serializable interface.
 func (p *Version) DecodeBinary(br *io.BinReader) {
 	p.Magic = config.NetMode(br.ReadU32LE())
 	p.Version = br.ReadU32LE()
-	p.Services = br.ReadU64LE()
 	p.Timestamp = br.ReadU32LE()
-	p.Port = br.ReadU16LE()
 	p.Nonce = br.ReadU32LE()
 	p.UserAgent = br.ReadVarBytes()
-	p.StartHeight = br.ReadU32LE()
-	p.Relay = br.ReadBool()
+	br.ReadArray(&p.Capabilities, capability.MaxCapabilities)
+	unique := make(map[capability.Type]bool)
```

Comment:
Three actual types, 32 maximum values, can we solve this problem without a map?

Reply:
Done this just with bool variables.